downstairs/src/region.rs

```diff
@@ -470,10 +470,21 @@ impl Extent {
             number,
             block_size: def.block_size(),
             extent_size: def.extent_size(),
-            inner: Mutex::new(Inner { file, metadb }),
+            inner: Some(Mutex::new(Inner { file, metadb })),
         })
     }
 
+    /**
+     * Close an extent and the metadata db files for it.
+     */
+    pub fn close(&mut self) -> Result<()> {
+        let inner = self.inner.as_ref().unwrap().lock().unwrap();
```

Comment:
This means a panic if the extent is already closed - we could match on the contents of Inner here, unless this panic is intentional :)

Reply:
Sounds good, resolving.